executor/executor_test.go

```diff
@@ -2959,3 +2959,66 @@ func (s *testSuite) TestUnionAutoSignedCast(c *C) {
 	tk.MustQuery("select id, v from t5 union select id, v from t7 union select id, v from t6 order by id").
 		Check(testkit.Rows("1 1", "2 -1", "3 -1"))
 }
+
+func (s *testSuite) TestUpdateJoin(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t1, t2, t3, t4, t5")
+	tk.MustExec("create table t1(k int, v int)")
+	tk.MustExec("create table t2(k int, v int)")
+	tk.MustExec("create table t3(id int auto_increment, k int, v int, primary key(id))")
+	tk.MustExec("create table t4(k int, v int)")
+	tk.MustExec("create table t5(k int, v int, primary key(k))")
+	tk.MustExec("insert into t1 values (1, 0)")
+	tk.MustExec("insert into t4 values (3, 3)")
+
+	// test auto_increment & none-null column in right table of update left join.
+	tk.MustExec("update t1 left join t3 on t1.k = t3.k set t1.v = 1")
+	tk.MustQuery("select k, v from t1").Check(testkit.Rows("1 1"))
+	tk.MustQuery("select id, k, v from t3").Check(testkit.Rows())
+
+	// test left join and right no records but update no records part.
+	tk.MustExec("update t1 left join t2 on t1.k = t2.k set t1.v = t2.v, t2.v = 3")
+	tk.MustQuery("select k, v from t1").Check(testkit.Rows("1 <nil>"))
+	tk.MustQuery("select k, v from t2").Check(testkit.Rows())
+
+	// test set right table and ref right old value in left one.
+	tk.MustExec("update t1 left join t2 on t1.k = t2.k set t2.v = 3, t1.v = t2.v")
+	tk.MustQuery("select k, v from t1").Check(testkit.Rows("1 <nil>"))
+	tk.MustQuery("select k, v from t2").Check(testkit.Rows())
+
+	// test right join and left no records but update no records part.
```

Comment:
-> test right join and the case that the left table has no matching record but has updated the left table columns